src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -52,9 +54,18 @@
   }
 
   private final boolean useCompactMapEntries;
+  private final BehaviorOnNullValues behaviorOnNullValues;
 
-  public DataConverter(boolean useCompactMapEntries) {
+  /**
+   * @param behaviorOnNullValues How to treat records with null values. Cannot be null; if in doubt,
+   * {@link BehaviorOnNullValues#DEFAULT} can be used.
+   */
+  public DataConverter(boolean useCompactMapEntries, BehaviorOnNullValues behaviorOnNullValues) {
     this.useCompactMapEntries = useCompactMapEntries;
+    this.behaviorOnNullValues = Objects.requireNonNull(
+        behaviorOnNullValues,
+        "behaviorOnNullValues cannot be null. Possible fix: use BehaviorOnNullValues.DEFAULT instead."
+    );
```

Comment:
IMO, the message is not necessary. This constructor is only called by the `ElasticsearchWriter` (and lots of tests), and the `behaviorOnNullValue` should be null only when a developer introduced a bug.

Reply:
Makes sense, will shorten the message to `"behaviorOnNullValues cannot be null"` (unless that's still too much and no message is preferable).

Reply:
Is that even required? Why not just `Objects.requireNonNull(behaviorOnNullValues)`?

Reply:
It isn't, just seems like it might make life slightly easier for anyone running into the NPE in the future. Message can definitely be removed as an argument if that's better.